fang/src/asynk/backend_sqlx.rs

```diff
@@ -0,0 +1,995 @@
+                )
+                .await?;
+
+                Ok(Res::Bigint(affected_rows))
+            }
+
+            Q::RemoveTask => {
+                let affected_rows =
+                    general_any_impl_remove_task(REMOVE_TASK_QUERY_MYSQL, pool, params).await?;
+
+                Ok(Res::Bigint(affected_rows))
+            }
+            Q::RemoveTaskByMetadata => {
+                let affected_rows = general_any_impl_remove_task_by_metadata(
+                    REMOVE_TASK_BY_METADATA_QUERY_MYSQL,
+                    pool,
+                    params,
+                )
+                .await?;
+
+                Ok(Res::Bigint(affected_rows))
+            }
+            Q::RemoveTaskType => {
+                let affected_rows =
+                    general_any_impl_remove_task_type(REMOVE_TASKS_TYPE_QUERY_MYSQL, pool, params)
+                        .await?;
+
+                Ok(Res::Bigint(affected_rows))
+            }
+            Q::FetchTaskType => {
+                let task =
+                    general_any_impl_fetch_task_type(FETCH_TASK_TYPE_QUERY_MYSQL, pool, params)
+                        .await?;
+                Ok(Res::Task(task))
+            }
+            Q::FindTaskById => {
+                let task: Task =
+                    general_any_impl_find_task_by_id(FIND_TASK_BY_ID_QUERY_MYSQL, pool, params)
+                        .await?;
+
+                Ok(Res::Task(task))
+            }
+            Q::RetryTask => {
+                let task = mysql_impl_retry_task(RETRY_TASK_QUERY_MYSQL, pool, params).await?;
+
+                Ok(Res::Task(task))
+            }
+            Q::InsertTaskIfNotExists => {
+                let task = mysql_any_impl_insert_task_if_not_exists(
+                    (
+                        FIND_TASK_BY_UNIQ_HASH_QUERY_MYSQL,
+                        INSERT_TASK_UNIQ_QUERY_MYSQL,
+                    ),
+                    pool,
+                    params,
+                )
+                .await?;
+
+                Ok(Res::Task(task))
+            }
+        }
+    }
+
+    fn _name() -> &'static str {
+        "MySQL"
+    }
+}
+
+async fn mysql_impl_insert_task(
+    query: &str,
+    pool: &Pool<Any>,
+    params: QueryParams<'_>,
+) -> Result<Task, AsyncQueueError> {
+    let uuid = Uuid::new_v4();
+    let mut buffer = Uuid::encode_buffer();
+    let uuid_as_str: &str = uuid.as_hyphenated().encode_lower(&mut buffer);
+
+    let scheduled_at_str = format!("{}", params.scheduled_at.unwrap().format("%F %T%.f+00"));
+
+    let metadata_str = params.metadata.unwrap().to_string();
+    let task_type = params.task_type.unwrap();
+
+    let affected_rows = sqlx::query(query)
+        .bind(uuid_as_str)
+        .bind(metadata_str)
+        .bind(task_type)
+        .bind(scheduled_at_str)
+        .execute(pool)
+        .await?
+        .rows_affected();
+
+    if affected_rows != 1 {
+        // here we should return an error
+        panic!("fock")
```

Comment:
yes, here we should create an error

Reply:
[return an error instead of panicking with fock](https://github.com/ayrat555/fang/pull/141/commits/e31296d702e8a5523514ebfdd7e1bdcad48d13aa)